vault/eventbus/bus_test.go

```diff
@@ -180,3 +184,177 @@ func TestBus2Subscriptions(t *testing.T) {
 		t.Error("Timeout waiting for event2")
 	}
 }
+
+// TestBusSubscriptionsCancel verifies that canceled subscriptions are cleaned up.
+func TestBusSubscriptionsCancel(t *testing.T) {
+	subscriptions.Store(0)
+	bus, err := NewEventBus(nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+	ctx := context.Background()
+	bus.Start()
+
+	// create and cancel a bunch of subscriptions
+	create := 100
+	cancel := 50
+
+	eventType := logical.EventType("someType")
+
+	var channels []<-chan *logical.EventReceived
+	var cancels []context.CancelFunc
+	stopped := atomic.Int32{}
```

Comment:
Seems unused? But also the better name if we tabulate this and the below test.

Reply:
Ah, left in from a previous iteration of this test :)